src/matlab2tikz.m

```diff
@@ -892,37 +892,111 @@ case guitypes()
 end
 % ==============================================================================
 function m2t = drawGridOfAxes(m2t, handle)
-    % draws the grids of an axes
-    %TODO: has{XYZ}Grid is always false without a good reason
-    hasXGrid = false;
-    hasYGrid = false;
-    hasZGrid = false;
-    if hasXGrid || hasYGrid || hasZGrid
-        matlabGridLineStyle = get(handle, 'GridLineStyle');
-        % Take over the grid line style in any case when in strict mode.
-        % If not, don't add anything in case of default line grid line style
-        % and effectively take Pgfplots' default.
-        defaultMatlabGridLineStyle = ':';
-        if m2t.cmdOpts.Results.strict ...
-                || ~strcmpi(matlabGridLineStyle,defaultMatlabGridLineStyle)
-            gls = translateLineStyle(matlabGridLineStyle);
-            axisGridOpts = {'grid style', sprintf('{%s}', gls)};
-            m2t.axesContainers{end}.options = cat(1, ...
-                m2t.axesContainers{end}.options,...
-                axisGridOpts);
+    % Draws the grids of an axis
+    options = opts_new();
+
+    % Check for major/minor grids
+    hasGrid = [isOn(get(handle, 'XGrid'));
+               isOn(get(handle, 'YGrid'));
+               isOn(get(handle, 'ZGrid')) && isAxis3D(handle)];
+
+    hasMinorGrid = [isOn(get(handle, 'XMinorGrid'));
+                    isOn(get(handle, 'YMinorGrid'));
+                    isOn(get(handle, 'ZMinorGrid')) && isAxis3D(handle)];
+
+    xyz = {'x', 'y', 'z'};
+
+    % Check for local grid options
+    for i=1:3
+        if hasGrid(i)
+            grid     = [xyz{i}, 'majorgrids'];
+            options = opts_add(options, grid, []);
         end
-    else
+        if hasMinorGrid(i)
+            grid     = [xyz{i}, 'minorgrids'];
+            options = opts_add(options, grid, []);
+        end
+    end
+
+    % Check for global grid options
+    if any(hasGrid)
+        gridOpts = [];
```

Comment:
This is not an options array. The problem is, that these global options *all* end up within a 
```
grid style = { }
```
Therefore, i gather the options in a string beforehand